tests/fertiscan/db/update_inspection/test_update_sub_labels.py

```diff
@@ -206,8 +207,84 @@ def test_update_sub_labels(self):
         for expected_updated_item in expected_updated_data:
             self.assertTrue(
                 expected_updated_item in updated_data,
-                f"Expected updated item {expected_updated_item} was not found in the updated data"
+                f"Expected updated item {expected_updated_item} was not found in the updated data",
+            )
+
+    def test_update_sub_labels_with_mismatched_arrays(self):
+        # Create sub labels with mismatched 'fr' and 'en' arrays
+        mismatched_sub_labels = json.dumps(
+            {
+                "instructions": {
+                    "fr": [
+                        "1. Dissoudre 50g dans 10L d'eau.",
+                        "2. Appliquer toutes les 2 semaines.",
+                    ],
+                    "en": [
+                        "1. Dissolve 50g in 10L of water.",
+                        "2. Apply every 2 weeks.",
+                        "3. Store in a cool, dry place.",
+                    ],
+                }
+            }
+        )
+
+        try:
+            # Execute the function with mismatched sub labels
+            self.cursor.execute(
+                "SELECT update_sub_labels(%s, %s);",
+                (self.label_id, mismatched_sub_labels),
             )
+        except Exception as e:
+            self.fail(f"update_sub_labels raised an exception unexpectedly: {e}")
+
+        # Verify that no exception was raised and check the data integrity
+        self.cursor.execute(
+            "SELECT text_content_fr, text_content_en FROM sub_label WHERE label_id = %s;",
+            (self.label_id,),
+        )
+        saved_data = self.cursor.fetchall()
+
+        # Expected data should still be saved even though the arrays were mismatched
+        expected_data = [
```

Comment:
I personally think it is enough to compare the number of sub_labels, no need to directly compare the content

Reply:
Also creating a `expected_data` feels weird, you should refer to your `mismatched_sub_labels` if you want to directly compare the content